x-pack/plugins/uptime/server/lib/adapters/monitors/elasticsearch_monitors_adapter.ts

```diff
@@ -187,35 +180,44 @@ export class ElasticsearchMonitorsAdapter implements UMMonitorsAdapter {
       },
     };
 
-    const queryResult = await this.database.search(request, params);
```

Comment:
This is great, I like the solution you chose for re-querying for the `after_key` case. I'd like to reduce (guffaw) the amount of nesting here, and I think we can eliminate a line or two in the process if we switch to a `do...while` loop instead.

What do you think of something like this:
```
    let searchAfter: any = null;
    const queryResults: Array<{ up: number; down: number }> = [];

    do {
      if (searchAfter) {
        set(params, 'body.aggs.ids.composite.after', searchAfter);
      }

      const queryResult = await this.database.search(request, params);
      const idBuckets = get(queryResult, 'aggregations.ids.buckets', []);

      const filterAndReduce = (accumulator: { up: number; down: number }, bucket: any) => {
        const check = get(bucket, 'latest.hits.hits[0]', undefined);
        const status: string = get(check, '_source.monitor.status');
        if (statusFilter && statusFilter !== status) {
          return accumulator;
        }
        if (status === 'up') {
          accumulator.up++;
        } else {
          accumulator.down++;
        }

        return accumulator;
      };

      queryResults.push(idBuckets.reduce(filterAndReduce, { up: 0, down: 0 }));

      searchAfter = get(queryResult, 'aggregations.ids.after_key', null);
    } while (searchAfter);

    const { up, down } = queryResults.reduce(
      (total, current) => ({
        up: total.up + current.up,
        down: total.down + current.down,
      }),
      { up: 0, down: 0 }
    );
```

Reply:
I just pushed a new commit with the `do {} while()` addition, I also changed more code to use `forEach` instead of `map` `reduce`. Would like your feedback on alternate approaches here, but it seemed to me that keeping the intermediate map results in memory was unnecessary when we only need to count.